### PR TITLE
ctypes: improve bytes handling

### PR DESCRIPTION
**Pull request by JelleZijlstra**

### CODE REVIEW

File: stdlib/ctypes/__init__.pyi [hunk 1]
```diff
@@ -194,7 +194,7 @@ class _SimpleCData(Generic[_T], _CData):
 class c_byte(_SimpleCData[int]): ...
 
 class c_char(_SimpleCData[bytes]):
-    def __init__(self, value: int | bytes = ...) -> None: ...
+    def __init__(self, value: int | bytes | bytearray = ...) -> None: ...
```
Comment (srittau) [inline stdlib/ctypes/__init__.pyi:197]: I failed to trace through the code for this, but I trust this is correct.

File: tests/stubtest_allowlists/py3_common.txt [hunk 2]
```diff
@@ -367,6 +367,8 @@ _ctypes.sizeof
 # Allowlist entries that cannot or should not be fixed
 # ==========
 
+ctypes.Array.raw  # exists but stubtest can't see it
```
Comment (srittau) [inline tests/stubtest_allowlists/py3_common.txt:370]: Nit: Maybe copy the comment from the stub?

**Review by srittau — APPROVED** [review]

File: tests/stubtest_allowlists/py3_common.txt (hunk 2 above)
Comment (JelleZijlstra) [inline tests/stubtest_allowlists/py3_common.txt:370]:
```suggestion
ctypes.Array.raw  # exists but stubtest can't see it; only available if _CT == c_char
```

File: stdlib/ctypes/__init__.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stdlib/ctypes/__init__.pyi:197]: I didn't find the code either but tried it in the REPL. This one allows bytearray but `c_char_p` doesn't.

**Pull request merged**